exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt

```diff
@@ -179,27 +178,59 @@ class JdbcDatabaseMetadataImpl(database: String, val metadata: DatabaseMetaData)
+     *
+     * - Not set: `varchar("any", 128).nullable()`
+     * - Set null: `varchar("any", 128).nullable().default(null)`
+     * - Set "NULL": `varchar("any", 128).nullable().default("NULL")`
+     * ```
+     * DB                  Not set    Set null                    Set "NULL"
+     * SqlServer           null       "(NULL)"                    "('NULL')"
+     * SQLite              null       "NULL"                      "'NULL'"
+     * Postgres            null       "NULL::character varying"   "'NULL'::character varying"
+     * PostgresNG          null       "NULL::character varying"   "'NULL'::character varying"
+     * Oracle              null       "NULL "                     "'NULL' "
+     * MySql5              null       null                        "NULL"
+     * MySql8              null       null                        "NULL"
+     * MariaDB3            "NULL"     "NULL"                      "'NULL'"
+     * MariaDB2            "NULL"     "NULL"                      "'NULL'"
+     * H2V1                null       "NULL"                      "'NULL'"
+     * H2V1 (MySql)        null       "NULL"                      "'NULL'"
+     * H2V2                null       "NULL"                      "'NULL'"
+     * H2V2 (MySql)        null       "NULL"                      "'NULL'"
+     * H2V2 (MariaDB)      null       "NULL"                      "'NULL'"
+     * H2V2 (PSQL)         null       "NULL"                      "'NULL'"
+     * H2V2 (Oracle)       null       "NULL"                      "'NULL'"
+     * H2V2 (SqlServer)    null       "NULL"                      "'NULL'"
+     * ```
+     * According to this table there is no simple rule of what is the default value. It should be checked
+     * for each DB (or groups of DBs) specifically.
+     * In the case of MySql and MariaDB it's also not possible to say whether was default value skipped or
+     * explicitly set to `null`.
+     *
+     * @return `null` - if the value was set to `null` or not configured. `defaultValue` in other case.
+     */
+    private fun sanitizedDefault(defaultValue: String): String? {
         val dialect = currentDialect
-        val h2Mode = dialect.h2Mode
         return when {
-            dialect is SQLServerDialect -> defaultValue.trim('(', ')', '\'')
-            dialect is OracleDialect || h2Mode == H2CompatibilityMode.Oracle -> defaultValue.trim().let {
-                if (it.startsWith('\'') && it.endsWith('\'')) it.trim('\'') else it
+            dialect is SQLServerDialect -> defaultValue.trim('(', ')').extractNullAndStringFromDefaultValue()
+            dialect is MariaDBDialect -> when (defaultValue) {
+                "NULL" -> null
+                else -> defaultValue.extractNullAndStringFromDefaultValue()
             }
-            dialect is MysqlDialect || h2Mode == H2CompatibilityMode.MySQL || h2Mode == H2CompatibilityMode.MariaDB -> defaultValue.substringAfter(
-                "b'"
-            ).trim('\'')
```

Comment:
I'm not sure if we have good enough test coverage to ok removing some of these branches. The h2Modes are probably not going to be an issue, but removing MariaDB from this substring check may cause problems in a few users.

One example I can think of:
In both MySQL and MariaDB, Exposed takes the opinion to store boolean values in `BOOLEAN` column type instead of `BIT`. But let's say a user has an existing table already in the DB that uses `BIT` (or chooses to create one themselves), and they are using their own custom `bool()` to register and work with this `BIT` column using Kotlin booleans.
Attempting something like the following would no longer show the same behavior in both DB:
```kt
// represents the existing table in db
exec("CREATE TABLE IF NOT EXISTS tester (bool BIT DEFAULT b'0' NOT NULL)")

// represents the identical Exposed definition
object TestTable : Table("tester") {
    val bool = bool("bool").default(false)
}

// when calling SchemaUtils.createMissingTables....
// MySQL - correctly does not generate ALTER statement
// MariaDB - incorrectly does generate ALTER statement
```

I'd suggest a third branch (if value starts with `b'` or something) in the new MariaDB logic that does what the original logic did (i.e. what the new MySQL branch does now). Not sure if the mysql H2 modes would also be affected.

Reply:
Added check of `'b` for MariaDB also, but without proper test actually.

But what I've found during testing of this is that `binary` column is almost untested and quite broken.
I added test to check binary column ddl generation, and insert/reading values (with binary literal also, and the binary literal actually). So now it looks better, but PR become a bit bigger..